MyApplication/app/src/main/java/net/serkanbal/wallmartlist/MainActivity.java

```diff
@@ -0,0 +1,118 @@
+
+        mButtonPS4.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getWallMartItemResults("PS4");
+            }
+        });
+
+        mButtonXboxOne.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getWallMartItemResults("Xbox+One");
+            }
+        });
+
+        mButtonWiiU.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getWallMartItemResults("Wii+U");
+            }
+        });
+    }
+
+    private void getWallMartItemResults(String query) {
+        OkHttpClient client = new OkHttpClient();
+
+        final Request request = new Request.Builder()
+                .url(GET_ITEMNAME+query)
+                .build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (!response.isSuccessful()) {
+                    throw new IOException("Failed: " + response);
+                } else {
+                    try {
+                        mList.clear();
+                        JSONObject newObj = new JSONObject(response.body().string());
+                        JSONArray newArray = newObj.getJSONArray("items");
+                        for (int i = 0; i < newArray.length(); i++) {
+                            JSONObject nameObj = newArray.getJSONObject(i);
+                            String actualName = nameObj.getString("name");
+                            mList.add(new ResultItem(actualName));
+
+                            MainActivity.this.runOnUiThread(new Runnable() {
+                                @Override
+                                public void run() {
+                                    mRecyclerView = (RecyclerView) findViewById(recyclerview);
+
+                                    LinearLayoutManager linearLayoutManager =
+                                            new LinearLayoutManager(MainActivity.this,
+                                            LinearLayoutManager.VERTICAL,false);
+
+                                    mRecyclerView.setLayoutManager(linearLayoutManager);
+
+                                    mAdapter =
+                                            new SearchResultRecyclerViewAdapter(mList);
+                                    mRecyclerView.setAdapter(mAdapter);
+
+                                    mAdapter.notifyDataSetChanged();
+
+                                }
+                            });
+                        }
+                    } catch (JSONException e) {
+                        e.printStackTrace();
```

Comment:
Let the user know something went wrong, otherwise they'll think your app just froze and didn't do anything